src/crimson/osd/osd.cc

```diff
@@ -246,12 +246,31 @@ seastar::future<> OSD::start()
     return seastar::when_all_succeed(
       cluster_msgr->try_bind(pick_addresses(CEPH_PICK_ADDRESS_CLUSTER),
                              local_conf()->ms_bind_port_min,
-                             local_conf()->ms_bind_port_max)
-        .then([this] { return cluster_msgr->start(&dispatchers); }),
+                             local_conf()->ms_bind_port_max
+      ).then([this] (bool ret) {
+        if (!ret) {
+          throw std::runtime_error(fmt::format(
+              "cluster_msgr cannot bind to {}, port range [{}, {}]",
+              pick_addresses(CEPH_PICK_ADDRESS_CLUSTER),
+              local_conf()->ms_bind_port_min,
+              local_conf()->ms_bind_port_max));
+        }
+        return cluster_msgr->start(&dispatchers);
+      }),
       public_msgr->try_bind(pick_addresses(CEPH_PICK_ADDRESS_PUBLIC),
                             local_conf()->ms_bind_port_min,
-                            local_conf()->ms_bind_port_max)
-        .then([this] { return public_msgr->start(&dispatchers); }));
+                            local_conf()->ms_bind_port_max
+      ).then([this] (bool ret) {
+        if (!ret) {
+          throw std::runtime_error(fmt::format(
```

Comment:
please note, this is not a "runtime_error". runtime_error is an error caused by runtime not the program itself.

Reply:
fixed, see https://github.com/ceph/ceph/pull/32632/commits/daa3cc2375369dcf0f9e5cbfdcded78e6a86b30e